tests/conftest.py

```diff
@@ -73,8 +75,46 @@ def capture_and_mock(*args, **kwargs):
 
             return MockResponse(response_dict, status_code, headers=headers)
 
-        monkeypatch.setattr(requests, method, capture_and_mock)
+        monkeypatch.setattr(requests, "request", capture_and_mock)
 
         return (request_args, request_kwargs)
 
     return inner
+
+
+@pytest.fixture
+def mock_pagination_request(monkeypatch):
```

Comment:
I don't like how complex this test function is, but we need to roughly emulate the behavior of a pagination endpoint and return different pages of data in successive calls.